src/dir.cr

```diff
@@ -282,6 +284,7 @@ class Dir
     include Iterator(String)
 
     def initialize(@dir : Dir)
+      @dir.rewind
```

Comment:
I think we should rewind in the first call to `next`, it's a bit weird just that creating the iterator rewinds the dir. But for that we need more state in the iterator, and turn it into a class. It's probably fine.

Reply:
As you wish.